app/grandchallenge/cases/serializers.py

```diff
@@ -219,6 +240,15 @@ def _get_linked_task(*, targets, interface):
         if interface:
             kwargs["interface_pk"] = interface.pk
         return add_images_to_archive.signature(kwargs=kwargs, immutable=True,)
+    elif "archive_item" in targets:
+        kwargs = {
+            "archive_item_pk": targets["archive_item"].pk,
+        }
+        if interface:
+            kwargs["interface_pk"] = interface.pk
```

Comment:
If the `interface_pk` is not set I think that `add_images_to_archive_item` will fail as-is (maybe I'm missing something). Is doing this covered by a test?

Reply:
Hmm, it does seem like it is covered in `try upload without interface`, but, I'm confused about why `ComponentInterface.objects.get(pk=interface_pk)` when `interface_pk = None` works.

Reply:
Correct, the interface does need to be set. If it is not set, a validation error is thrown during validation of the serlizalier, so `add_images_to_archive_item` never gets executed unless an interface is defined. That's why the test works. But I have adjusted the function definition accordingly now to reflect that the interface needs to be set, my bad for overlooking this!

Reply:
This is what code review is for, don't worry!